exercises/collatz-conjecture/example.rkt

```diff
@@ -0,0 +1,23 @@
+#lang racket
+
+(provide collatz)
+
+; uses a hash table to speed up subsequent calls
+(define (collatz-length ht n)
+  (cond 
+    [(hash-ref ht n #f) => (lambda (colval) colval)]
+    [else
+     (define colval 0)
+     (if (even? n) 
+         (set! colval (add1 (collatz-length ht (/ n 2))))
+         (set! colval (add1 (collatz-length ht (add1 (* 3 n))))))
```

Comment:
This seems unnecessarily redundant, why not collapse the repeated code and move the `(if (even? n) ...)` to where it matters?

Reply:
It would be even better to make `collatz-length` tail-recursive. Perhaps that obviates the need for the memoization. What led you to use a hash table anyway?